### ISSUE TITLE
Allow ModuleRef to get imported providers

### ISSUE THREAD

**Issue by Discountrobot** [nestjs/nest#832]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## What is the motivation / use case for changing the behavior?
We have a have a generic controller that needs to resolve requests based off on some context specific parameters. these context specific implementations are contained in their own module and expected to implement and export an abstract with a token known to the client.

This is currently working by using the `@Inject()` decorator. However we would like to know nothing about context specific modules inside the generic implementation, with the obvious exception of the module declaration.

The ideal solution to this would be ModuleRef behaving identical to the normal DI. (#258)
or otherwise allowing for lookups in both internal and external providers or by extending `.get` or adding an additional method.

Which allows us to have cleaner modularity, and might allow for ideas like inclusion/exclusion of modules during build time.

I'll gladly provide a PR for this feature, should it be accepted (:

## Current behavior

```typescript
const LION_TOKEN = 'LION_TOKEN';
const CAT_TOKEN = 'CAT_TOKEN';

const lionProvider = {
    provide: CAT_TOKEN,
    useValue: 'Meow',
};

const catProvider = {
    provide: LION_TOKEN,
    useValue: 'Roar!',
};

@Module({
    providers: [lionProvider],
    exports: [lionProvider],
})
class LionModule {}

@Module({
    providers: [catProvider],
    exports: [catProvider],
})
class CatModule {}

@Module({
    imports: [LionModule, CatModule],
})
class FelineModule {

    constructor(
        @Inject(CAT_TOKEN) private readonly catVoice: string,
        @Inject(LION_TOKEN) private readonly lionVoice: string
    ) {
        console.log(this.getFelineVoice(LION_TOKEN));
    }

    getFelineVoice(token: string): string {
        switch (token) {
            case CAT_TOKEN:
                return this.catVoice;
            case LION_TOKEN:
                return this.lionVoice;
            default:
                throw new HttpException('unsupported feline', HttpStatus.BAD_REQUEST);
        }
    }

}
```
## Expected behavior

```typescript
@Module({
    imports: [LionModule, CatModule],
})
class FelineModule {

    constructor(
        private readonly moduleRef: ModuleRef
    ) {
        console.log(this.getFelineVoice(LION_TOKEN));
    }

    getFelineVoice(token: string): string {
        const feline = this.moduleRef.get<string>(token);

        if (feline == null) {
            throw new HttpException('unsupported feline', HttpStatus.BAD_REQUEST);
        }

        return feline;
    }

}
```

## Environment

<pre><code>
Nest version: 5.0.1

</code></pre>

**Comment by kamilmysliwiec** [comment]
This feature will be available once #903 is merged:
```javascript
this.moduleRef.get<string>(token, { strict: false });
```
By default `strict` is set to `true`, which means that `ModuleRef` is looking for provider only within the current context

**Comment by kamilmysliwiec** [comment]
5.2.0 is here!
